Week-6/Problem_6.6/sum.js

```diff
@@ -0,0 +1,29 @@
+function sumThree(arr, element) {
+  let sum = 0;
+  let length = arr.length;
+  for (let i = 0; i < length; i++) {
+    for (let j = i + 1; j < length; j++) {
+      for (let k = j + 1; k < length; k++) {
+        // sum of three elements
+        let sum3 = arr[i] + arr[j] + arr[k];
+        // checking the absolute difference and comparing
+        if (Math.abs(element - sum) >= Math.abs(element - sum3)) {
+          sum = sum3;
+        }
+      }
+    }
+  }
+  return sum;
+}
+
```

Comment:
1. Sort the given array.
2. Loop over the array and fix the first element of the possible triplet, arr[i].
3. Then fix two pointers, one at (i + 1) and the other at n – 1. And look at the sum,
    1. If the sum is smaller than the sum we need to get to, we increase the first pointer.
    2. Else, If the sum is bigger, Decrease the end pointer to reduce the sum.
    3. Update the closest sum found so far.